skbio/stats/composition.py

```diff
@@ -807,6 +807,319 @@ def centralize(mat):
+def _vlr(x:np.array, y:np.array, ddof:int):
+    r"""
+    Calculates variance log ratio
+    
+    Parameters
+    ----------
+    x : array_like, float
+       a 1-dimensional vector of proportions
+    y : array_like, float
+       a 1-dimensional vector of proportions
+        
+    ddof: int
+        degrees of freedom
+        
+       
+    Returns
+    -------
+    float
+         variance log ratio value
+         
+    
+    References
+    ----------
+    .. [1] V. Lovell D, Pawlowsky-Glahn V, Egozcue JJ, Marguerat S, Bähler J (2015) 
+           Proportionality: A Valid Alternative to Correlation for Relative Data. 
+           PLoS Comput Biol 11(3): e1004075. https://doi.org/10.1371/journal.pcbi.1004075
+    .. [2] Erb, I., Notredame, C. 
+           How should we measure proportionality on relative gene expression data?. 
+           Theory Biosci. 135, 21–36 (2016). https://doi.org/10.1007/s12064-015-0220-8
+    """
+
+    # Log transformation
+    x = np.log(x)
+    y = np.log(y)
+    
+    #Variance log ratio
+    return np.var(x - y, ddof=ddof)
+
+
+@experimental(as_of="0.5.7")
+def _robust_vlr(x:np.array, y:np.array, ddof:int):
+    r"""
+    Calculates variance log ratio while masking zeros
+    
+    Parameters
+    ----------
+    x : array_like, float
+       a 1-dimensional vector of proportions
+    y : array_like, float
+       a 1-dimensional vector of proportions
+        
+    ddof: int
+        degrees of freedom
+        
+    Returns
+    -------
+    float
+         variance log ratio value
+         
+    
+    References
+    ----------
+    .. [1] V. Lovell D, Pawlowsky-Glahn V, Egozcue JJ, Marguerat S, Bähler J (2015) 
+           Proportionality: A Valid Alternative to Correlation for Relative Data. 
+           PLoS Comput Biol 11(3): e1004075. https://doi.org/10.1371/journal.pcbi.1004075
+    .. [2] Erb, I., Notredame, C. 
+           How should we measure proportionality on relative gene expression data?. 
+           Theory Biosci. 135, 21–36 (2016). https://doi.org/10.1007/s12064-015-0220-8
+    """
+
+    # Mask zeros
+    x = np.ma.masked_array(x, mask=x == 0)
+    y = np.ma.masked_array(y, mask=y == 0)
+    
+    # Log transformation
+    x = np.ma.log(x)
+    y = np.ma.log(y)
+    
+    #Variance log ratio
+    return np.ma.var(x - y, ddof=ddof)
+
+@experimental(as_of="0.5.7")
+def vlr(x:np.array, y:np.array, ddof:int=1, robust:bool=False):
+    r"""
+    Calculates variance log ratio
+    
+    Parameters
+    ----------
+    x : array_like, float
+       a 1-dimensional vector of proportions
+    y : array_like, float
+       a 1-dimensional vector of proportions
+        
+    ddof: int
+        degrees of freedom
+        
+    robust: bool
+        mask zeros at the cost of performance
+       
+    Returns
+    -------
+    float
+         variance log ratio value
+         
+    Examples
+    --------
+    No zeros
+    >>> x = [1,2,3]
+    >>> y = [5,8,13]
+    >>> %timeit vlr(x,y)
```

Comment:
let's cut out the timings out of the docstrings - they probably won't work there anyways

Reply:
I thought I removed all of them.  That one must have slipped through the cracks.